### DISCUSSION TITLE
how to add and fill a TabPane to an existing TabbedContent

### DISCUSSION THREAD

**Discussion by nnako** [Textualize/textual#5291] [Q&A] [answered]
In the [demo application](https://github.com/Textualize/textual/blob/main/src/textual/demo/widgets.py), the [widget description](https://github.com/Textualize/textual/blob/main/docs/widgets/tabbed_content.md) and the [api documentation](https://textual.textualize.io/widgets/tabbed_content/#textual.widgets.TabbedContent), `TabbedContent` elements are always created using the `with` statement and a couple of `yield` statements to add respective child `TabPane` elements.

Unfortunately, within my own application, I was not able to add **and define** a new `TabPane` element to an existing `TabbedContent` element. Within the api documentation, I am led to the method `add_pane()` of the `TabbedContent` class. And within this method, I can reference a `TabPane` object. But I don't know how to add a valid `*children` element of the latter. As I understand it, these `*children` elements would hold the content for the respective new tab:

```python
profiles = [ 'Tab1', 'Tab2', 'Tab3' ]
for _profile in profiles:
    _TabPane = _TabbedContent.add_pane(
        pane=TabPane(
            title=_profile,
            name=_profile,
            id=_profile.lower(),

?           children=Static("<description>", classes="profile",)

            )
        )

```

Is there a way to dynamically add and fill further `TabPane` elements? There has been a [solution](https://github.com/Textualize/textual/discussions/2412) presented for a similar question, but that one from 2023 would definitely be too verbose and might not even work anymore.

Thanks.

**Comment by TomJGooding** [comment, marked as answer]
It looks like you are pretty much there already, unless I've misunderstood your question.

Here's a quick example:

```python
from textual.app import App, ComposeResult
from textual.widgets import Footer, Static, TabbedContent, TabPane


class ExampleApp(App):
    BINDINGS = [
        ("a", "add_panes", "Add panes"),
    ]

    def compose(self) -> ComposeResult:
        yield TabbedContent()
        yield Footer()

    def action_add_panes(self) -> None:
        tabbed_content = self.query_one(TabbedContent)
        for i in range(1, 4):
            tabbed_content.add_pane(
                TabPane(
                    f"Tab{i}",
                    Static(f"Tab{i} content"),
                )
            )


if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```

**Reply by willmcgugan** [reply]
Good solution. You can also subclass `TabPane` and yield its children from `compose()`.

**Comment by nnako** [comment]
Thanks

My "mistake" was to use the parameters of `TabPane` as **keyword arguments** and not try to use them as **positional arguments**.

As this seems to be a general concept for documentation, where do I find notes about how to use a parameter which is [written](https://textual.textualize.io/widgets/tabbed_content/#textual.widgets.TabPane) as e.g. `*children`? Reading it within the documentation, I would try to use a proper keyword. But the solution would be to "forget" the given keyword and provide the parameter as a positional one.

**Reply by willmcgugan** [reply]
The `*` syntax denotes positional arguments in Python. https://docs.python.org/3/tutorial/controlflow.html#arbitrary-argument-lists
